modules/core/src/main/java/org/apache/ignite/client/ClientCache.java

```diff
@@ -293,11 +529,36 @@
      */
     public boolean putIfAbsent(K key, V val) throws ClientException;
 
+    /**
+     * Atomically associates the specified key with the given value if it is not already associated with a value.
+     * <p>
+     * This is equivalent to:
+     * <pre><code>
+     * if (!cache.containsKey(key)) {}
+     *   cache.put(key, value);
+     *   return true;
+     * } else {
+     *   return false;
+     * }
+     * </code></pre>
+     * except that the action is performed atomically.
+     *
+     * @param key Key with which the specified value is to be associated.
+     * @param val Value to be associated with the specified key.
+     * @return <tt>true</tt> if a value was set.
+     */
+    public IgniteClientFuture<Boolean> putIfAbsentAsync(K key, V val) throws ClientException;
+
     /**
      * Clears the contents of the cache.
      */
     public void clear() throws ClientException;
 
+    /**
+     * Clears the contents of the cache.
```

Comment:
Probably, needs a `@return` tag and the information about how this is different from `removeAllAsync`

Reply:
Fixed